toolchain.py

```diff
@@ -814,19 +794,6 @@ def build(self, arch):
         self.delete_marker("building")
         self.set_marker("build_done")
 
-    def update_state(self, key, value):
```

Comment:
Oh man so that method was duplicated :facepalm: :smile: 
How did we survive that long without static analysis and unit testing :sweat_smile: